File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.py
def solution(a, b):
    gcd = 1
    
    for i in range(a, 0, -1):
        if a % i == 0 and b % i == 0:
            gcd = i
            break
            
    b /= gcd
    while b % 2 == 0: b /= 2
    while b % 5 == 0: b /= 5
    
    return 1 if b == 1 else 2

File: 프로그래머스/0/120878. 유한소수 판별하기/유한소수 판별하기.js
function solution(a, b) {
    let gcd = 0;
    
    for (let i = a; i >= 1; i--) {
        if (a % i == 0 && b % i == 0) {
            gcd = i;
            break;
        }
    }
    
    b /= gcd;
    while (b % 2 == 0) b /= 2;
    while (b % 5 == 0) b /= 5;
    
    return b == 1 ? 1 : 2;
}